notebooks/official/migration/UJ1 Vertex SDK AutoML Image Classification.ipynb

```diff
@@ -0,0 +1,1432 @@
+{
```

Comment:
<p>1) I don't think you need to use <code>try-catch</code>  here. In fact, using try-catch will prevent any errors from being caught by the testing pipeline.</p><p>The user will still see the error message if try-catch is removed.</p><p> </p><p>2) What is the <code>in globals()</code>  check for? Recommend removing unless there's a good reason.</p><p> </p><p>3) All the other notebooks set <code>delete_bucket</code>  to False so we the user doesn't accidentally delete their entire bucket by accident.</p><p>I know that this means GCS files are not cleaned up, but we can rely on the scheduled cleaner for that (although it currently doesn't do this).</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/33/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='86'/>

Reply:
It's a catch all cleanup (from reuse repos)
1. The globals() is to verify that the object was created, before deleting it.
2. try/catch may be a legacy from earlier GAPIC days when things could still throw an exception.
For example, let's say cleanup deletes the dataset, but the notebook already deleted the dataset. In this case, I think the delete might throw an exception.

Reply:
It seems a contradiction that the default is to cleanup all generated resources, but not the temporary bucket.

Reply:
Yes, you could say that they might have other models that this would inadvertently delete.

I would say that, to be safe, either:
1) Delete only specific resources created by this notebook
2) Delete everything (models, endpoints, etc) but hide behind a single flag that is set to False.

If you agree, I'll make the change to all the other notebooks.

Reply:
I will revisit the design of cleanup

Reply:
updated